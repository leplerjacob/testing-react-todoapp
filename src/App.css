.App {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.header.welcome {
  color: #fcc34d;
}
