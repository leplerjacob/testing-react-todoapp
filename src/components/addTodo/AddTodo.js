import { useState } from "react";
import "./addtodo.css";

export default function AddTodo(props) {
  const [todo, setTodo] = useState("");
  const [date, setDate] = useState("");

  const addTodo = props.addTodo;

  function changeValue(e) {
    switch (e.target.name) {
      case "todo": {
        setTodo(e.target.value);
      }
      case "date": {
        setDate(e.target.value);
      }
      default: {
        return;
      }
    }
  }

  function createTodoAndSend() {
    const createdTodo = {
      id: null,
      todo: todo,
      date: date,
    };
    addTodo(createdTodo);
  }

  return (
    <form>
      <label htmlFor="todo">To Do</label>
      <input
        name="todo"
        type="text"
        id="input todo"
        value={todo}
        onChange={changeValue}
      />
      <label htmlFor="date">Date</label>
      <input
        name="date"
        type="date"
        id="input date"
        value={date}
        onChange={changeValue}
      />
      <label htmlFor="submit">Submit</label>
      <input type="button" value="Add Todo" onClick={createTodoAndSend} />
    </form>
  );
}
