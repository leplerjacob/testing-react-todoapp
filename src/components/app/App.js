import React, { useState } from "react";
import "./App.css";
import AddTodo from "../addTodo/AddTodo";

const initialTodos = [
  { id: 1, todo: "Walk the dog", date: "2021-10-20" },
  { id: 2, todo: "Visit grandparents", date: "2021-10-22" },
  { id: 3, todo: "Attend React conference", date: "2021-10-26" },
];

function App() {
  const [todos, setTodos] = useState(initialTodos);

  function addTodo(todo) {
    setTodos([...initialTodos, todo])
  }

  return (
    <div className="App">
      <h1 className="header welcome">Todo Application</h1>
      <AddTodo addTodo={addTodo} />
    </div>
  );
}

export default App;
