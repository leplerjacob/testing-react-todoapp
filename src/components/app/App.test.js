import React from "react";
import { render, screen } from "@testing-library/react";
import Enzyme, { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import App from "./App";
import AddTodo from "../addTodo/AddTodo";

Enzyme.configure({ adapter: new Adapter() });

describe("App properties", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  afterEach(() => {
      wrapper.unmount()
  })

  test("renders app component", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("app renders todoApp component", () => {
    expect(wrapper.find(AddTodo)).toHaveLength(1);
  });

  test("app has initial state", () => {
      console.log(wrapper.state("todos"))
  });

  test("app state changes after event", () => {});
});
