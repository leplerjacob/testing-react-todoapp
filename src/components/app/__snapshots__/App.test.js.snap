// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App properties renders app component 1`] = `
<div
  className="App"
>
  <h1
    className="header welcome"
  >
    Todo Application
  </h1>
  <AddTodo
    addTodo={[Function]}
  />
  <ListTodos
    todos={
      Array [
        Object {
          "date": "2021-10-20",
          "id": 1,
          "todo": "Walk the dog",
        },
        Object {
          "date": "2021-10-22",
          "id": 2,
          "todo": "Visit grandparents",
        },
        Object {
          "date": "2021-10-26",
          "id": 3,
          "todo": "Attend React conference",
        },
      ]
    }
  />
</div>
`;
